modules/http/tests/http_validation.cc

```diff
@@ -37,8 +47,107 @@ BOOST_AUTO_TEST_CASE(test_parse_url)
 	testGString.allocated_len=8;
 	testGString.len=3;
 	strcpy(testPlainString,"GET");
-	proxyFake.request_url=g_string_new("http://example.com");
-	proxyFake.request_method=&testGString;
-	http_split_request(&proxyFake,line,33);
+	proxyFake->request_method=&testGString;
+
+	http_split_request(proxyFake,line,33);
 	BOOST_CHECK(testPlainString[8]=='b');
 }
+
+typedef struct {
+	const char* input_line;
+	const char* expected_method;
+	const char* expected_url;
+	const char* expected_version;
+} TestCase;
+
+TestCase testCases[] = {
```

Comment:
I'd name this `validRequests` so it would be paired with the failing (`invalid`?) ones.